src/message.ts

```diff
@@ -1111,17 +1112,55 @@ export class MediaMessage extends Message {
       if (!this.obj.url) {
         throw new Error('no url')
       }
-      log.verbose('MediaMessage', 'stream() url: %s', this.obj.url)
+      log.verbose('MediaMessage', 'readyStream() url: %s', this.obj.url)
       return UtilLib.urlStream(this.obj.url, cookies)
     } catch (e) {
-      log.warn('MediaMessage', 'stream() exception: %s', e.stack)
+      log.warn('MediaMessage', 'readyStream() exception: %s', e.stack)
       Raven.captureException(e)
       throw e
     }
   }
 
-  public forward(room: Room): Promise<boolean>
-  public forward(contact: Contact): Promise<boolean>
+  /**
+   * save file
+   *
+   * @param filePath save file
+   */
+  public async saveFile(filePath: string): Promise<boolean> {
+    let ret = false
+    if (!filePath) {
+      throw new Error('saveFile() filePath is invalid')
+    }
+    log.silly('MediaMessage', `saveFile() filePath:'${filePath}'`)
+    if (fs.existsSync(filePath)) {
+      throw new Error('saveFile() file does exist!')
+    }
+    const writeStream = fs.createWriteStream(filePath)
+    let readStream
+    try {
+      readStream = await this.readyStream()
+    } catch (e) {
+      log.error('MediaMessage', `saveFile() call readyStream() error: ${e.message}`)
+      throw new Error(`saveFile() call readyStream() error: ${e.message}`)
+    }
+    ret = <boolean>await new Promise((resolve, reject) => {
+      readStream.pipe(writeStream)
+      readStream
+        .on('end', () => {
+          log.silly('MediaMessage', `saveFile() pipe end`)
+          resolve(true)
```

Comment:
I'd like to suggest move `resolve()` to the following code:
```ts
writeStream.once('close', resolve)
```
and also add this to monitor failure event:
```ts
writeStream.once('error', reject)
```